### PR TITLE
Add token tests

### PR DESCRIPTION
**Pull request by MikeHathaway**
- Switches proxy from transparent proxy to the UUPS proxy standard defined in ERC1967 (best practice recommendation)
- Implements proxy in tests and executes tests against the proxy
- Adds OZ base library required for proxy contract instantiation

Future TODOS:

- Split token and proxy into separate contracts
- split token functionality and proxy functionality into separate test contracts

### CODE REVIEW

File: test/Token.t.sol [hunk 1]
```diff
@@ -1,12 +1,118 @@
...
+        // check different implementation addresses
+        assertTrue(address(_token) != address(tokenV2));
+
+        // check new method can be accessed and correctly reads state
+        assertEq(tokenProxyV2.testVar(), 0);
+        tokenProxyV2.setTestVar(100);
+        assertEq(tokenProxyV2.testVar(), 100);
+
+        // TODO: check previous state no longer accessible...
+        // assertEq(tokenProxyV1.totalSupply(), 1_000_000_000 * 10 ** tokenProxyV1.decimals());
```
Comment (EdNoepel) [inline test/Token.t.sol:140]: Assume that'll be in a future PR.

**Review by EdNoepel — APPROVED** [review]
LGTM

**Comment by MikeHathaway** [comment]
Discussion of different proxy types availabe in docs here: https://docs.openzeppelin.com/contracts/4.x/api/proxy#transparent-vs-uups

File: test/Token.t.sol (hunk 1 above)
Comment (MikeHathaway) [reply test/Token.t.sol:140]: Yes, I'm currently working on testing upgrading and transferring contract administration which I'll make a separate PR for.

**Pull request merged**